crates/bevy_ecs/src/bundle.rs

```diff
@@ -14,120 +14,185 @@ use bevy_ecs_macros::all_tuples;
-/// }
-/// ```
+/// Adding a [`Bundle`] value to an entity will add the components from the set it
+/// represents to the entity.
+/// The values of these components are taken from the [`Bundle`].
+/// If an entity already had one of these components, the entity's original component value
+/// will be overwritten.
+///
+/// Importantly, `Bundle`s have no meaning outside of their constituent set of components.
+/// You **should not** use bundles as a unit of behaviour.
+/// The behaviour of your app can only be considered in terms of the interaction
+/// between systems and the bundle's constituent components.  
+/// This rule is important because multiple bundles may contain the same component - if bundles
+/// were treated as an abstraction boundary, this would be an implementation leakage.
+///
+/// For this reason, `bevy` does not store which bundles were used to create a given entity.
+/// There is also intentionally no [`Query`] filter which detects whether an entity contains the components
+/// of a bundle.
+/// To select multiple components in a query, you should specify the components exactly.
+///
+/// ## Removal
+///
+/// Bundles can also be used to remove components from an entity.
+///
+/// Removing a bundle from an entity will remove the components of the set it
+/// represents from the entity.
+/// A value of the [`Bundle`] type may then be collected containing these component values,
+/// and returned.
+///
+/// # Implementors
+///
+/// A [`Component`] can be added to or removed from a world.
+/// Therefore, every type which implements [`Component`] also implements [`Bundle`].
+///
+/// Additionally, [Tuples](`tuple`) of bundles are also [`Bundle`] (with up to 15 bundles).
+/// These bundles contain the items of the 'inner' bundles.
+/// This is a convenient shorthand which is primarily used when spawning entities.
+/// For example, spawning an entity using the bundle `(SpriteBundle {...}, PlayerMarker)`
+/// will spawn an entity with components required for a 2d sprite, and the `PlayerMarker` component.
+///
+/// [`unit`], otherwise known as [`()`](`unit`), is a [`Bundle`] containing no components (since it
+/// can also be considered as the empty tuple).
+/// This can be useful for spawning large numbers of empty entities using
+/// [`World::spawn_batch`](crate::world::World::spawn_batch)
+///
+/// Tuple bundles can be nested, which can be used to create an anonymous bundle with more than
+/// 15 items.
+/// However, in most cases where this is required, there is a [`derive@Bundle`] derive instead.
+/// This is used to create your own bundles from a struct of other [`Bundle`]s.
+/// As explained above, this includes any [`Component`] type, and other derived bundles:
 ///
-/// You can nest bundles using the `#[bundle]` attribute:
 /// ```
 /// # use bevy_ecs::{component::Component, bundle::Bundle};
 ///
 /// #[derive(Component)]
-/// struct X(i32);
-/// #[derive(Component)]
-/// struct Y(u64);
+/// struct XPosition(i32);
```

Comment:
Calling these like `PositionX` or, more shortly, `PosX` (I somehow like more the former) is easier to the eyes IMO.

Reply:
Perhaps? It's incongruent with the order in any other uses I've seen of similar terms.
I'd rather leave it like this

Happy for you to suggest an alternative scenario however.